src/main/include/log4cxx/spi/location/locationinfo.h

```diff
@@ -45,21 +58,28 @@ class LOG4CXX_EXPORT LocationInfo
 
 		static const LocationInfo& getLocationUnavailable();
 
-
+#ifdef LOG4CXX_HAS_STRING_VIEW
+		static constexpr const char* calcShortFileName(const char* fileName){
+			std::string_view view(fileName);
+			// If the separator is not found, rfind will return -1.  Adding 1 to
+			// that will have it pointing at fileName, which is a good fallback.
+			return fileName + view.rfind(LOG4CXX_SHORT_FILENAME_SPLIT_CHAR) + 1;
+		}
+#else
+		static const char* calcShortFileName(const char* fileName){
+			return strrchr(fileName, LOG4CXX_SHORT_FILENAME_SPLIT_CHAR) + 1;
```

Comment:
`strrchr` might return a NULL pointer in case SPLIT_CHAR is not found [according to the docs](https://www.cplusplus.com/reference/cstring/strrchr/?kw=strrchr). Shouldn't that be handled? You explicitly mention that case for `string_view`. `NULL + 1` seems wrong.

Reply:
whoops you're right.  If it's found we need to add 1(like the string_view version), else we should return the `fileName` as a sane default value.